### PR TITLE
Fixed python extension build

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Fixes for building wheel files on linux/docker

### CODE REVIEW

File: pulsar-client-cpp/docker/build-wheels.sh [hunk 1]
```diff
@@ -44,5 +44,5 @@ for line in "${PYTHON_VERSIONS[@]}"; do
     IMAGE_NAME=$BUILD_IMAGE_NAME:manylinux-$PYTHON_SPEC
 
     echo "Using image: $IMAGE_NAME"
-    docker run -i -v $PWD:/pulsar $IMAGE_NAME /build-wheel-file-within-docker.sh
+    docker run -i -v $PWD:/pulsar $IMAGE_NAME /pulsar/pulsar-client-cpp/docker/build-wheel-file-within-docker.sh
```
Comment (ivankelly) [inline pulsar-client-cpp/docker/build-wheels.sh]: nit: -i is unnecessary

File: pulsar-client-cpp/docker/build-wheel-file-within-docker.sh [hunk 2]
```diff
@@ -18,19 +18,22 @@
 # under the License.
 #
 
-set -e
+set -e -x
```
Comment (ivankelly) [inline pulsar-client-cpp/docker/build-wheel-file-within-docker.sh]: nit: can be done in one switch "set -ex"

File: pulsar-client-cpp/docker/build-wheel-file-within-docker.sh [hunk 3]
```diff
@@ -18,19 +18,22 @@
...
+        -DBUILD_TESTS=OFF
 
 make clean
-make _pulsar
+make _pulsar -j3
 
 cd python
 python setup.py bdist_wheel
+
+auditwheel repair dist/pulsar_client-*-$PYTHON_SPEC-linux_x86_64.whl
```
Comment (ivankelly) [inline pulsar-client-cpp/docker/build-wheel-file-within-docker.sh]: a comment on why auditwheel is needed would be useful.

File: pulsar-client-cpp/docker/build-wheels.sh (hunk 1 above)
Comment (merlimat) [reply pulsar-client-cpp/docker/build-wheels.sh]: Wouldn't the container run in background without the `-i`? We need to wait for the script to complete.

File: pulsar-client-cpp/docker/build-wheel-file-within-docker.sh (hunk 3 above)
Comment (merlimat) [reply pulsar-client-cpp/docker/build-wheel-file-within-docker.sh]: Sure, will add

File: pulsar-client-cpp/docker/build-wheels.sh (hunk 1 above)
Comment (ivankelly) [reply pulsar-client-cpp/docker/build-wheels.sh]: that would be -d. -i just keeps stdin open.

**Pull request merged**